src/cfnlint/rules/resources/s3/BucketPolicy.py

```diff
@@ -0,0 +1,181 @@
+                    matches.extend(
+                        self._check_statement(
+                            statement, statement_path[:],
+                            bucket_resources, custom_resources))
+
+        return matches
+
+    def _check_statement(self, value, path, bucket_resources, custom_resources):
+        """ Check each statement"""
+        matches = []
+
+        actions = value.get('Action')
+        if not isinstance(actions, (list, six.string_types)):
+            return matches
+        resources = value.get('Resource')
+        if not isinstance(resources, (list, six.string_types, dict)):
+            return matches
+
+        if isinstance(actions, six.string_types):
+            actions = [actions]
+        if isinstance(resources, (six.string_types, dict)):
+            resources = [resources]
+
+        found_object_actions, found_bucket_actions, actions_results_conclusive = self._get_statement_actions(actions)
+        found_object_resources, found_bucket_resources, bucket_results_conclusive = self._get_statement_resources(
+            resources, bucket_resources, custom_resources)
+        if not actions_results_conclusive or not bucket_results_conclusive:
+            return matches
+
+        if (not found_object_resources) and found_object_actions:
+            message = 'S3 Bucket Policy has object actions and no specified object resource'
+            matches.append(RuleMatch(path[:] + ['Resource'], message))
+        elif not found_bucket_resources and found_bucket_actions:
+            message = 'S3 Bucket Policy has bucket actions and no specified bucket resource'
+            matches.append(RuleMatch(path[:] + ['Resource'], message))
+
+        return matches
+
+    def _get_statement_actions(self, actions):
+        """ Breaks down the actions in a statement to each category """
+        # print(self.s3_actions)
+        object_actions = list(filter(re.compile('.*Object.*').match, self.s3_actions))
+        object_actions.extend(self.s3_object_actions)
+
+        bucket_actions = ['s3:' + action for action in list(set(self.s3_actions) - set(object_actions))]
+        object_actions = ['s3:' + action for action in object_actions]
+
+        found_object_actions = []
+        found_bucket_actions = []
+        for action in actions:
+            if action in object_actions:
+                found_object_actions.append(action)
+            elif action in bucket_actions:
+                found_bucket_actions.append(action)
+            else:
+                return None, None, False
+
+        return found_object_actions, found_bucket_actions, True
+
+    def _get_statement_resources(self, resources, bucket_resources, custom_resources):
+        """ Get Statement Resources """
+        found_bucket_resources = False
+        found_object_resources = False
+        for resource in resources:
+            if isinstance(resource, dict):
+                if len(resource) == 1:
```

Comment:
Only 1 resource? Is this true? What if:

```
- Resource:
  - "bucketA"
  - "bucketB/object"
```

🤔

Reply:
In this case I think this is right.  Its because it can be 
- `Resource: !Sub "arn:aws:s3:::${myBucket}"`
- `Resource: !Ref myBucket` 
etc

Reply:
That being said I don't think I covered lists very well so I may have to add some logic there.